Spigot-API-Patches/0130-isChunkGenerated-API.patch

```diff
@@ -0,0 +1,60 @@
+From 1b4fe89e169ba05876718fe662b42f1dae61678e Mon Sep 17 00:00:00 2001
+From: cswhite2000 <18whitechristop@gmail.com>
+Date: Tue, 21 Aug 2018 19:39:46 -0700
+Subject: [PATCH] isChunkGenerated API
+
+Resolves #1329
+
+diff --git a/src/main/java/org/bukkit/Location.java b/src/main/java/org/bukkit/Location.java
+index 7e1ee875..48b4f56e 100644
+--- a/src/main/java/org/bukkit/Location.java
++++ b/src/main/java/org/bukkit/Location.java
+@@ -502,6 +502,14 @@ public class Location implements Cloneable, ConfigurationSerializable {
+     public boolean isChunkLoaded() { return world.isChunkLoaded(locToBlock(x) >> 4, locToBlock(z) >> 4); } // Paper
+ 
+     // Paper start
++    /**
++     * Checks if a {@link Chunk} has been generated at this location.
++     *
++     * @return true if a chunk has been generated at this location
++     */
++    public boolean isGenerated() {
++        return world.isChunkGenerated(locToBlock(x) >> 4, locToBlock(z) >> 4);
```

Comment:
`world` can be null, so it might be good idea to throw in a check from Preconditions?

Reply:
It seems that the methods in `Location` that utilize `world` are inconsistent about checking whether or not world is null. If we should check to see if `world` is null here, than we should also change the rest of the methods in `Location` that utilize world, so that they also check if  `world` is null.

Reply:
Good point about the inconsistency, but imo following the dumb decisions done previously isn't mandatory. You are here to make the API better, you can do it properly.

I mean, isn't it nice to see anything better than `java.lang.NullPointerExceptiom` without a message?

Reply:
I really would love to understand why someone allowed world to be null in the first dang place for this class :/